File: 0152-maximum-product-subarray/0152-maximum-product-subarray.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int n=nums.size();
        int maxp=INT_MIN;;
        for(int start=0;start<n;start++){
            int product=1;
            for(int end=start;end<n;end++){
                product*=nums[end];
                  maxp=max(product,maxp);
            }
           
        }
        return maxp;

        // using kadane's algorithm

        // int maxp=nums[0];
        // int curr_max=nums[0];
        // int curr_min=nums[0];
        // for(int i=1;i<n;i++){
        //     if
            
        // }

    }
};

File: 0152-maximum-product-subarray/0152-maximum-product-subarray.java
class Solution {
    public int maxProduct(int[] nums) {
        int max=nums[0];
        int maxp=nums[0];
        int minp=nums[0];
        for(int i=1;i<nums.length;i++){
           if(nums[i]<0){
            int temp=maxp;
            maxp=minp;
            minp=temp;
           }
            maxp=Math.max(nums[i],maxp*nums[i]);
            minp=Math.min(nums[i],minp*nums[i]);
            max=Math.max(maxp,max);
        }
        return max;
    }
}